### PR TITLE
Feature - Update aws-toolkit-commons package

### PR DESCRIPTION
**Pull request by damntrecky**
- update AmazonQ Code Transform telemetry definitions for new package update

## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
We update the AWS Toolkit Common package and need to update the VSCode package per this PR -> https://github.com/aws/aws-toolkit-common/pull/648

## Checklist
- [x ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]

File: .changes/next-release/bugfix-32139a16-d00e-43e2-aab8-a356008abfd1.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Fix telemetry logging for new Amazon Q Code Transform telemetry updates"
```
Comment (rli) [inline .changes/next-release/bugfix-32139a16-d00e-43e2-aab8-a356008abfd1.json:3]: non-userfacing so does not need a changelog

Comment (damntrecky) [reply .changes/next-release/bugfix-32139a16-d00e-43e2-aab8-a356008abfd1.json:3]: Thanks for letting me know. Should I delete?

Comment (rli) [reply .changes/next-release/bugfix-32139a16-d00e-43e2-aab8-a356008abfd1.json:3]: happy to keep, doesn't hurt

**Review by smoring2 — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt [hunk 2]
```diff
@@ -136,7 +136,7 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
                 false,
                 message("codemodernizer.notification.warn.invalid_project.description.reason.missing_content_roots"),
                 InvalidTelemetryReason(
-                    CodeTransformPreValidationError.EmptyProject
+                    CodeTransformPreValidationError.NoPom
```
Comment (tincheng) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:139]: Is it possible to keep the reason more generic, such as NoBuildFile? Q Transformation will support other buildTools like Gradle and "missing_content_roots" is applicable to these other buildTools.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt [hunk 3]
```diff
@@ -160,7 +160,7 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
                 false,
                 message("codemodernizer.notification.warn.invalid_project.description.reason.no_valid_files", supportedBuildFileNames.joinToString()),
                 InvalidTelemetryReason(
-                    CodeTransformPreValidationError.ProjectSelectedIsNotJava8OrJava11,
+                    CodeTransformPreValidationError.NonMavenProject,
```
Comment (tincheng) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:163]: Once we started supporting other build tools like Gradle, this telemetry would likely become something more generic like `NoSupportedProjectType` if project is neither Maven or Gradle. Would it be better to start using that?

**Review by arangatang — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt (hunk 2 above)
Comment (damntrecky) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:139]: Right now its not clear what we are going to support. And right now specifics around maven/gradle are useful metrics to have to determine how many projects we have or do not have with our specific requirements.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt (hunk 3 above)
Comment (damntrecky) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:163]: Right now specifics around maven/gradle are useful metrics to have to determine how many projects we have or do not have with our specific requirements.

**Pull request merged**